File: Python/Easy/3146.py
class Solution:
    def findPermutationDifference(self, s: str, t: str) -> int:
        # use a hashmap to store the index position of a character
        temp = {}
        answer = 0
        # fill the dictionary
        for i in range(len(s)):
            temp[s[i]] = i

        # compare with the other string
        for i in range(len(t)):
            holder = abs(temp[t[i]] - i)
            answer += holder

        return answer

File: Java/Easy/3146.java
class Solution {
    public int findPermutationDifference(String s, String t) {
        int sum = 0;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            int indexT = t.indexOf(c);
            sum += Math.abs(i - indexT);
        }
        return sum;
    }
}